proposal/nested-map-functions.md

```diff
@@ -0,0 +1,389 @@
+//     'hover': (
+//       'search': green,
+//       'home': red,
+//       'filter': blue,
+//       'logo': orange,
+//     ),
+//   ),
+// );
+```
+
+## Functions
+
+All new and modified functions are part of the `sass:map` built-in module.
+
+### `get()`
+
+This proposal updates the signature and behavior of the existing `get()`
+function.
+
+> This also affects the global `map-get()` function.
+
+```
+get($map, $key, $keys...)
+```
+
+> Intuitively, `get($map, $key1, $key2, $key3)` is equivalent to
+> `get(get(get($map, $key1), $key2), $key3)` with the exception that if any
+> intermediate value isn't a map or doesn't have the given key the whole
+> function returns `null` rather than throwing an error.
+
+* If `$map` is not a map, throw an error.
+
+* Let `child` be `$map`.
+
+* Let `keys` be a list containing `$key` followed by the elements of `$keys`.
+
+* For each element `key` in `keys`:
+
+  * If `child` is not a map, return `null`.
+
+  * If `child` contains a key that's `==` to `key`, set `child` to the value
+    associated with that key. Otherwise, return `null`.
+
+* Return `child`.
+
+### `has-key()`
+
+This proposal updates the signature and behavior of the existing `get()`
+function.
+
+> This also affects the global `map-has-key()` function.
+
+```
+has-key($map, $key, $keys...)
+```
+
+> Intuitively, `has-key($map, $key1, $key2, $key3)` is equivalent to
+> `has-key(get(get($map, $key1), $key2), $key3)` with the exception that if any
+> intermediate value isn't a map or doesn't have the given key the whole
+> function returns `false` rather than throwing an error.
+
+* If `$map` is not a map, throw an error.
+
+* Let `child` be `$map`.
+
+* Let `keys` be a list containing `$key` followed by the elements of `$keys`.
+
+* For each element `key` in `keys`:
+
+  * If `child` is not a map, return `false`.
+
+  * If `child` contains a key that's `==` to `key`, set `child` to the value
+    associated with that key. Otherwise, return `false`.
+
+* Return `true`.
+
+### `set()`
+
+> Note: For consistency with other functions whose multi-key overloads were
+> added after their single-key versions, `set()` is defined to have a separate
+> single-key overload and multi-key overload.
+
+* ```
+  set($map, $key, $value)
+  ```
+
+  > Intuitively, `set($map, $key, $value)` is equivalent to `merge($map, ($key: $value))`.
+
+  * If `$map` is not a map, throw an error.
+
+  * Let `map` be a copy of `$map`.
+
+  * If `map` has a key that's `==` to `$key`, remove it and its associated value.
+
+  * Associate `$key` with `$value` in `map`.
+
+  * Return `map`.
+
+* ```
+  set($map, $args...)
+  ```
+
+  > Intuitively, `set($map, $key1, $key2, $value)` is equivalent to `set($map,
+  > $key1, set(get($map, $key1), $key2, $value))` with the exception that if any
+  > intermediate value isn't set or isn't a map it's replaced with a map.
+
+  * If `$map` is not a map, throw an error.
+
+  * If `$args` has fewer than three elements, throw an error.
```

Comment:
Should this still work with two elements?

Reply:
Either way, it's never going to be called with two elements because the prior overload takes priority. Given that, I think it's a little clearer to make its behavior totally disjoint with that overload's.